scripts/aem/aem.sh

```diff
@@ -0,0 +1,328 @@
+                   -w /home/trenchboot/grub \
+                   -e HOME=/home/trenchboot/grub/.home \
+                   --user "$(id -u):$(id -g)" \
+                   ghcr.io/trenchboot/trenchboot-sdk:master /bin/bash -c '
+            set -e
+            [ ! -f configure ] && ./bootstrap
+            [ ! -d build ] && (
+                mkdir build && cd build
+                ../configure --disable-werror --prefix=/ --disable-nls
+            )
+            make -C build -j$(nproc)
+            make -C build install DESTDIR=$PWD/build-install
+        '
+        if [ $? -ne 0 ]; then
+            echo 'Failed to build GRUB'
+            return 1
+        fi
+    fi
+
+    if [ "$what" = all ] || [ "$what" = xen ]; then
+        print_banner 'Building Xen'
+        docker run --rm -it \
+                   -v "$PWD/xen:/home/trenchboot/xen" \
+                   -w /home/trenchboot/xen \
+                   -e HOME=/home/trenchboot/xen/.home \
+                   --user "$(id -u):$(id -g)" \
+                   ghcr.io/trenchboot/trenchboot-sdk:master \
+                   make -j$(nproc) build-xen
+        if [ $? -ne 0 ]; then
+            echo 'Failed to build Xen'
+            return 1
+        fi
+    fi
+}
+
+function action_purge() {
+    kill_server
+
+    rm -rf grub xen
+    rm -rf webroot/grub webroot/xen
+    rm -f webroot/aem-bootstrap.sh webroot/grub.files webroot/index.html
+}
+
+function action_help() {
+    print_usage
+}
+
+function action_init() {
+    if ! git clone "$grub_repo" -b "$grub_branch" "$@" grub; then
+        echo 'Failed to clone GRUB'
+        return 1
+    fi
+    if ! git clone "$xen_repo" -b "$xen_branch" "$@" xen; then
+        echo 'Failed to clone Xen'
+        return 1
+    fi
+}
+
+function esc() {
+    printf '%q' "$@"
+}
+
+function validate_files_to_send() {
+    for entry in "${files_to_send[@]}"; do
+        IFS=":" read -r file dest_dir <<< "$entry"
+        if [ ! -f "webroot/$file" ]; then
+            error "Invalid \$files_to_send entry: '$entry'?"
+            error "  Make sure webroot/$file file (not directory!) exists."
+        fi
+    done
+
+    if [ -n "$should_die" ]; then
+        error 'Aborting, adjust "aem-cfg.sh" file in current directory' \
+              'or add missing files.'
+        exit 1
+    fi
+}
+
+function build_webroot() {
+    local grub_install=$PWD/grub/build-install
+
+    mkdir -p webroot
+
+    rm -rf webroot/grub
+    mkdir webroot/grub/
+    ln -s \
+       "$grub_install"/sbin/grub-install \
+       "$grub_install"/lib/grub/i386-pc/* \
+       webroot/grub/
+
+    mkdir -p webroot/xen/
+    ln -sf "$PWD/xen/xen/xen.gz" webroot/xen
+
+    find webroot/grub -type l |
+        sed "s#^webroot/#$server_url/#" > webroot/grub.files
+
+    cat > webroot/aem-bootstrap.sh <<EOF
+#!/bin/bash
+
+set -e
+
+mnt=/tmp/aes-mnt
+mkdir -p "\$mnt"
+
+# in case previous run of the script hasn't finished for some reason
+umount -q "\$mnt" || true
```

Comment:
`-q` isn't available in all `umount` versions. When it isn't available, no unmounting takes place and full usage is printed. Redirecting output to `/dev/null` may be safer.

Reply:
Thanks.  Second `umount` didn't have `-q`.  I updated the line in a copy outside of the repo, tested and forgot to apply the change.